packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -230,12 +231,43 @@ class Dart2JSTarget extends Dart2WebTarget {
   };
 
   @override
-  List<String> get buildFiles => <String>[
+  Iterable<File> buildFiles(Environment environment)
+    => environment.buildDir
+      .listSync(recursive: true)
+      .whereType<File>()
+      .where((File file) {
+        if (file.basename == 'main.dart.js') {
+          return true;
+        }
+        if (file.basename == 'main.dart.js.map') {
+          return compilerConfig.sourceMaps;
+        }
+        final RegExp partFileRegex = RegExp(r'main\.dart\.js_[0-9].*\.part\.js');
```

Comment:
Is there a risk of a silent failure if `dart2js` changes the file structure but we forget to update these regexes? Is there a way to fail hard during a Dart SDK roll?

Reply:
I think that would require some larger end-to-end integration test. I'm going to file an issue as a follow-on for adding an integration test for this.

Reply:
https://github.com/flutter/flutter/issues/146359